llama_stack/distribution/start_container.sh

```diff
@@ -90,7 +90,6 @@ $DOCKER_BINARY run $DOCKER_OPTS -it \
   $env_vars \
   -v "$yaml_config:/app/config.yaml" \
   $mounts \
-  $docker_image:$version_tag \
-  python -m llama_stack.distribution.server.server \
-  --yaml-config /app/config.yaml \
-  --port "$port"
+  --env PORT=$port \
```

Comment:
Hm, how is this environment variable read? I don't think there's a place where we read PORT?

Reply:
`llama_stack.distribution.server.server` can read this. You can pass either via flag or env var. I picked to use an env var to avoid having to escape the quotes within `--entrypoint`

```
usage: server.py
       [-h]
       [--yaml-config YAML_CONFIG]
       [--template TEMPLATE]
       [--port PORT]
       [--disable-ipv6]
       [--env ENV]
```

Reply:
Hold on. I don't think this works as I expected earlier. Let me double check.

Reply:
@ashwinb Ok I had to get it from env vars explicitly. Verified that this worked. I prefer this approach other than escaping from a complicated `--entrypoint`.

Reply:
I wonder if we could name this `LLAMASTACK_PORT` perhaps since there are a bunch of ports hanging around typically

Reply:
Great idea. Updated to `LLAMASTACK_PORT`.